.github/workflows/pr-builder.yaml

```diff
@@ -1,9 +1,17 @@
 on:
   workflow_call:
+    inputs:
+      needs:
+        required: false
+        type: string
+        default: '{}'
 
 jobs:
   run:
     runs-on: ubuntu-latest
     steps:
       # This reusable workflow should depend on all other jobs in the calling workflow.
-      - run: exit 0
+      - name: "Check all dependent jobs"
+        env:
+          NEEDS: ${{ inputs.needs }}
+        run: echo "$NEEDS" | jq -e 'map(select((.result == "success" or .result == "skipped") | not)) | length == 0'
```

Comment:
If I knew how, I'd write this as `any(result in {"success", "skipped"} for result in results)`. Specifically it'd be nice to short-circuit on the first non-match and use a set/list/array of acceptable statuses.

Reply:
Done